chapter1/README.md

```diff
@@ -109,86 +111,130 @@ There are two projects in this sample. Each needs to be registered separately in
    - In the **Name** section, enter a meaningful application name that will be displayed to users of the app, for example `TodoListSPA`.
-   - Under **Supported account types**, select **Accounts in this organizational directory**.
-   - In the **Redirect URI** section, select **Single-page application** in the combo-box and enter the following redirect URI: `http://localhost:4200/`.
+   - Under **Supported account types**, select **Accounts in this organizational directory only**.
+   - In the **Redirect URI (optional)** section, select **Single-page application** in the combo-box and enter the following redirect URI: `http://localhost:4200/`.
 1. Select **Register** to create the application.
 1. In the app's registration screen, find and note the **Application (client) ID**. You use this value in your app's configuration file(s) later in your code.
-1. In the app's registration screen, select **Authentication** in the menu.
-   - If you don't have a platform added, select **Add a platform** and select the **Single-page application** option.
-   - In the **Implicit grant** section, check the **Access tokens** and **ID tokens** option as this sample requires
-     the [Implicit grant flow](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-implicit-grant-flow) to be enabled to
-     sign-in the user, and call an API.
-
 1. Select **Save** to save your changes.
-1. In the app's registration screen, click on the **API permissions** blade in the left to open the page where we add access to the Apis that your application needs.
-   - Click the **Add a permission** button and then,
+1. In the app's registration screen, select the **API permissions** blade in the left to open the page where we add access to the APIs that your application needs.
+   - Select the **Add a permission** button and then,
    - Ensure that the **My APIs** tab is selected.
    - In the list of APIs, select the API `TodoListAPI`.
-   - In the **Delegated permissions** section, select the **access_as_user** in the list. Use the search box if necessary.
-   - Click on the **Add permissions** button at the bottom.
+   - In the **Delegated permissions** section, select the **Access 'TodoListAPI'** in the list. Use the search box if necessary.
+   - Select the **Add permissions** button at the bottom.
+
+#### Define Application Roles
+
+1. Still on the same app registration, select the **App roles** blade to the left.
+1. Select **Create app role**:
+    - For **Display name**, enter a suitable name, for instance **TaskAdmin**.
+    - For **Allowed member types**, choose **User**.
+    - For **Value**, enter **TaskAdmin**.
+    - For **Description**, enter **Admins can read any user's todo list**.
+1. Select **Create app role**:
+    - For **Display name**, enter a suitable name, for instance **TaskUsers**.
+    - For **Allowed member types**, choose **User**.
+    - For **Value**, enter **TaskUsers**.
```

Comment:
Value should be **TaskUser** as per SPA config file.

Reply:
done!